iaso/api/mobile/entity_type.py

```diff
@@ -75,18 +81,47 @@ def get_serializer_class(self):
         return MobileEntityTypeSerializer
 
     def get_queryset(self):
+        app_id = self.request.query_params.get("app_id")
+        user = self.request.user
+
+        queryset = EntityType.objects.filter(account=user.iaso_profile.account)
+
+        if not app_id:
+            raise ParseError("app_id is required")
+
+        try:
+            project = Project.objects.get_for_user_and_app_id(user, app_id)
+
+            if project.account is None and (not user or not user.is_authenticated):
+                raise AuthenticationFailed(f"Project Account is None or User not Authentified for app_id {app_id}")
+
+            queryset = queryset.filter(account=project.account)
 
-        queryset = EntityType.objects.filter(account=self.request.user.iaso_profile.account)
+        except Project.DoesNotExist:
+            if not user or not user.is_authenticated:
+                raise NotFound(f"Project Not Found and User not Authentified for app_id {app_id}")
```

Comment:
It's not a 404, it's a 401. It's a 404 is the user is authenticated though. Here, the exception is swallowed.